File: src/net/eduard/chat/command/ToggleTellCommand.java

package net.eduard.chat.command;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import net.eduard.api.lib.manager.CommandManager;
import net.eduard.chat.EduChat;

public class ToggleTellCommand extends CommandManager {


	
	 
	public ToggleTellCommand() {
		super("toggletell");
	}

	@Override
	public boolean onCommand(CommandSender sender, Command command,
			String label, String[] args) {
		if (sender instanceof Player) {
			Player p = (Player) sender;
			if (EduChat.getInstance().getChat().getTellDisabled().contains(p)) {
				EduChat.getInstance().getChat().getTellDisabled().remove(p);
				p.sendMessage("§cVocê desativou mensagens privadas");
			}else {
				EduChat.getInstance().getChat().getTellDisabled().add(p);
				p.sendMessage("§aVocê ativou mensagens privadas");
			}
			
		}
		return true;
	}


}

File: src/net/eduard/chat/command/ToggleTellCommand.kt
package net.eduard.chat.command

import net.eduard.api.lib.game.FakePlayer
import net.eduard.api.lib.manager.CommandManager
import net.eduard.chat.EduChat
import net.eduard.chat.core.ChatMessages
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class ToggleTellCommand : CommandManager("toggletell") {
    init{
        description= "Ativar ou deativar mensagens privadas"
    }
    override fun onCommand(sender: CommandSender, command: Command,
                           label: String, args: Array<String>): Boolean {
        if (sender is Player) {
            val p = sender
            val fake= FakePlayer(p)
            if (!EduChat.instance.chat.tellDisabled.contains(fake)) {
                EduChat.instance.chat.tellDisabled.add(fake)
                p.sendMessage(ChatMessages.tellOff)
            } else {
                EduChat.instance.chat.tellDisabled.remove(fake)
                p.sendMessage(ChatMessages.tellOn)
            }
        }
        return true
    }
}